### PR TITLE
Create Issue and Feature Request Templates for External Developers

### PR DESCRIPTION
**Pull request by sjanczy2**
### Background

We will be requesting external developers(partner and merchant devs) to open github issues in the argo repository for issues & feedback they have on checkout extensions. 

### Solution

To make this easier for them, I have created a few templates that they can use.

@lemonmade @denislins @jffortier Added you as reviewers but feel free to add others!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+about: Suggest an idea for this project
+title: ''
+labels: 'enhancement'
+assignees: ''
+
+---
+**Please list the related package(s)**
+e.g. argo-checkout, argo-checkout-react
+
+**If this is related to specific APIs or components, please list them here**
```
Comment (lemonmade) [inline .github/ISSUE_TEMPLATE/feature-request.md:12]: I like the way Polaris's issue template does this a bit better, with proper headings and HTML comments for the body so they never show up in the rendered issue: https://raw.githubusercontent.com/Shopify/polaris-react/main/.github/ISSUE_TEMPLATE/ISSUE.md

**Review by lemonmade — APPROVED** [review]
These seem like the right things to start with 👍 Left a suggestion to switch to Polaris's style of headings + comments blocks, which I think is a bit nicer, but I'm fine to ship with whatever and just iterate on it as we see what people do.

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 1 above)
Comment (sjanczy2) [reply .github/ISSUE_TEMPLATE/feature-request.md:12]: Good point, will update the samples later today. Thanks for sharing the Polaris examples!

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+title: ''
+labels: 'enhancement'
+assignees: ''
+
+---
+**Please list the related package(s)**
+e.g. argo-checkout, argo-checkout-react
```
[the remark this answers is not in the document]
Comment (sjanczy2) [reply .github/ISSUE_TEMPLATE/feature-request.md]: Done!

**Pull request merged**